### PR TITLE
Collaborator add/edit modal drawer

### LINKED ISSUE #166: Move add/edit contributor forms to a drawer
Most up-to-date design: https://xd.adobe.com/view/7c59a2ef-de0e-4296-a073-4ba7d551d5a2-290a/screen/f81695b1-dc2d-4d15-9e97-b048624117c9/

### PR DESCRIPTION
**Pull request by mbertrand**
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [ ] Mobile width screenshots
  - [x] Tag @ferdi or @pdpinch for review
- [ ] Testing
  - [ ] Code is tested
  - [ ] Changes have been manually tested

#### What are the relevant tickets?
Closes #166 

#### What's this PR do?
Moves the add/edit contributor forms to a modal drawer

#### How should this be manually tested?
Go to the collaborators page.  Add, edit, and delete collaborators.  All 3 should work as expected without errors.

#### Screenshots (if appropriate)

Screen Shot 2021-04-05 at 12 20 00 PM

Screen Shot 2021-04-05 at 12 20 12 PM

### CODE REVIEW

**Comment by mbertrand** [comment]
@ferdi @abdulkdawson screenshots are attached to the description.

**Comment by Ferdi** [comment]
Is the drop down same component as what George introduced in the other pr (react-select?)

**Comment by mbertrand** [comment]
@Ferdi the dropdown has not been modified, it is not using react-select.

File: static/js/components/SiteCollaboratorDrawer.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+          {collaborator ? `Edit ${collaborator.email}` : "Add collaborator"}
+        </ModalHeader>
+        <ModalBody>
+          <Card>
+            <div className="form-container m-3 p-3">
+              <SiteCollaboratorForm
+                collaborator={collaborator}
+                onSubmit={onSubmit}
+                onCancel={onCancel}
+              ></SiteCollaboratorForm>
```
Comment (gsidebo) [inline static/js/components/SiteCollaboratorDrawer.tsx:104]: Nit: this tag can be collapsed

File: static/js/components/SiteCollaboratorList.tsx [hunk 2]
```diff
@@ -64,75 +72,78 @@ export default function SiteCollaboratorList(): JSX.Element | null {
...
+                              className="edit-link"
+                              onClick={startEdit(collaborator)}
+                            >
+                              <i className="material-icons">edit</i>
+                            </a>
+                            <i
+                              className="material-icons"
+                              onClick={() => {
+                                setSelectedCollaborator(collaborator)
+                                toggleDeleteModal()
```
Comment (gsidebo) [inline static/js/components/SiteCollaboratorList.tsx:118]: I know this is existing code, but we have the `startEdit` helper method for running all the actions related to editing a collaborator. Seems appropriate that we'd have the same for deleting

**Review by gsidebo — APPROVED** [review]
Looks great! 👍  I left a couple very minor comments to take or leave

**Comment by Ferdi** [comment]
@mbertrand I think we should switch using react select for consistency .  This or other PR.

**Comment by mbertrand** [comment]
@Ferdi there is at least one other form that doesn't use react-select (the site creation form) so I think it would be best to tackle them all as a separate issue/PR.  I can start on that one next.

**Comment by mbertrand** [comment]
@gsidebo I made the suggested changes if you want to take a look at them.

**Review by gsidebo — APPROVED** [review]
👍 👍

**Pull request merged**